sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/FileTypeSensorTest.java

```diff
@@ -0,0 +1,188 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import java.io.File;
+import java.nio.file.Path;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.fs.internal.DefaultInputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+public class FileTypeSensorTest {
+  public static final String LANG_KEY = "LANG_KEY";
+
+  @Rule
+  public LogTester logTester = new LogTester();
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  private Path workDir;
+
+  private Settings settingsMock;
+  private SensorContextTester tester;
+  private ProjectTypeCollector projectTypeCollectorMock;
+  private DotNetPluginMetadata pluginMetadataMock;
+  private FileTypeSensor sensor;
+
+  @Before
+  public void prepare() throws Exception {
+    workDir = temp.newFolder().toPath();
+    settingsMock = mock(Settings.class);
+    tester = SensorContextTester.create(new File("src/test/resources"));
+    tester.fileSystem().setWorkDir(workDir);
+    tester.setSettings(settingsMock);
+
+    projectTypeCollectorMock = mock(ProjectTypeCollector.class);
+
+    pluginMetadataMock = mock(DotNetPluginMetadata.class);
+    when(pluginMetadataMock.languageKey()).thenReturn(LANG_KEY);
+
+    sensor = new FileTypeSensor(projectTypeCollectorMock, pluginMetadataMock);
+  }
+
+
+  @Test
+  public void should_describe() {
+
+    SensorDescriptor desc = mock(SensorDescriptor.class);
+    sensor.describe(desc);
+
+    verify(desc).name("Verify what types of files (MAIN, TEST) are in LANG_KEY projects.");
+  }
+
+  @Test
+  public void whenProjectNameNotPresent_shouldNotAddProjectInfo() {
+    when(settingsMock.getString("sonar.projectName")).thenReturn(null);
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs()).isEmpty();
+    verifyZeroInteractions(projectTypeCollectorMock);
+  }
+
+  @Test
+  public void whenLanguageKeyIsPresent_logsCsOutputFile() {
```

Comment:
Cs was CSharp originally?
```suggestion
  public void whenLanguageKeyIsPresent_logsOutputFile() {
```

Reply:
yes, initially I had 2 tests, one for `cs` and one for `vbnet` - refactoring leftover